[ci] Update CUDA toolkit to v12.1.1

antiagainst opened the pull request:
CUDA 12 enables users to target architecture-specific features in Hopper and Ada Lovelace architectures. It also enhances Graphs API.

Fixes https://github.com/openxla/iree/issues/13830

pzread commented on build_tools/docker/dockerfiles/nvidia.Dockerfile:
Line 55 also installs CUDA toolkit. We might need to also update it?

pzread commented on build_tools/docker/dockerfiles/nvidia-bleeding-edge.Dockerfile:
Same in this file on line 64, we also install CUDA toolkit there.

antiagainst committed:
[ci] Update CUDA toolkit to v12.1.1

CUDA 12 enables users to target architecture-specific features in
Hopper and Ada Lovelace architectures. It also enhances Graphs API.

Fixes https://github.com/openxla/iree/issues/13830
Closes https://github.com/openxla/iree/pull/13872

GMNGeoffrey reviewed (approved):
Some people (like me) don't have an Nvidia GPU on their dev machine, but still build the cuda target and driver. I think we use a forked parse_redist.py somewhere to fetch it during the build. Can you verify that this still works?

antiagainst committed:
Additionally install cuda_cccl for #include <nv/target>
